envoy/access_log/access_log.h

```diff
@@ -55,6 +56,34 @@ class AccessLogManager {
 
 using AccessLogManagerPtr = std::unique_ptr<AccessLogManager>;
 
+struct AccessLogTypeStringValues {
+  const std::string NotSet = EMPTY_STRING;
+  const std::string TcpUpstreamConnected = "TcpUpstreamConnected";
+  const std::string TcpPeriodic = "TcpPeriodic";
+  const std::string TcpEnd = "TcpEnd";
+  const std::string DownstreamStart = "DownstreamStart";
+  const std::string DownstreamPeriodic = "DownstreamPeriodic";
+  const std::string DownstreamEnd = "DownstreamEnd";
+  const std::string UpstreamStart = "UpstreamStart";
+  const std::string UpstreamPeriodic = "UpstreamPeriodic";
+  const std::string UpstreamEnd = "UpstreamEnd";
+};
+
+using AccessLogTypeStrings = ConstSingleton<AccessLogTypeStringValues>;
+
+enum class AccessLogType {
+  NotSet,
+  TcpUpstreamConnected,
+  TcpPeriodic,
+  TcpEnd,
+  DownstreamStart,
+  DownstreamPeriodic,
+  DownstreamEnd,
+  UpstreamStart,
+  UpstreamPeriodic,
+  UpstreamEnd,
+};
```

Comment:
Could we use the proto enum directly and just create an alias for it by `using`?

Reply:
done